### DISCUSSION TITLE
Have some questions about with FWI and RTM for OBN data

### DISCUSSION THREAD

**Discussion by zhangxiaoshuotttt** [slimgroup/JUDI.jl#177] [General]
Hi,
First of all, thanks for your help @mloubout @ziyiyin97 @kerim371, I processed the real OBN data with JUDI. But I still have some questions to ask you.

1. When I was going to do FWI with OBN data, I ran into a problem about the **elevation** of the **OBN nodes**. I notice that my nodes have **different elevations**, By reading the [references,](https://www.geoazur.fr/WIND/bin/view/Main/FWItuto/WebHome) I noticed that the computation of the sensitivity matrix relies on  the spatial reciprocity of the Green functions. Since I am new to FWI, **I do not know whether JUDI can deal with the FWI problem of receivers with such elevation changes.**Here is my original model.
![Starting OBN](https://user-images.githubusercontent.com/84370046/227517032-b549966c-efe9-44d3-a31c-7d6677a655f8.svg)

And here is my FWI result.
![Student](https://user-images.githubusercontent.com/84370046/227516787-c2d88993-dc7c-490c-92e4-35e15f78639c.svg)

2. In addition, I noticed that the coordinates of the sources and receivers of your model or open source data are all on the same line, But my source line and the receiver line are not on the same line, in other words, **my data is not 2D**, although I picked a source line that approximately coincides with the detector line, By means of projection, I approximately projected the source coordinates on the detection point line, But it looks like my RTM is not working well.  This is the result I obtained by directly performing RTM on the source line
![RTM_Student](https://user-images.githubusercontent.com/84370046/227517681-e7bb52b6-efdd-42a0-b8d0-7c18c1f4818c.svg)

So I used RTM for a Common Image Gather(I used this CIG line to obtain good imaging results by picking up the NMO velocity during the traditional prestack time migration process, after correcting for anisotropy and removing mutiples, But please forgive me I can't upload the imaging images). But it doesn't look good either.
![Student By CMP Gather](https://user-images.githubusercontent.com/84370046/227517852-737dde96-0ff6-4d4e-bdb2-aa0ea7faf31f.svg)
Here is my CIG:
![CIG](https://user-images.githubusercontent.com/84370046/227519862-2b282a2f-8db1-4435-b81a-5dacd224f353.png)

In order to solve the problem of real OBN data processing, I have been reading some papers recently, and I would also like to ask you for advice on dealing with OBN data where the coordinates of the receiver points are not on the same line (In other words, this is 3D data, although I only want to image one line at the moment), and the elevation of the detection points is fluctuated.

**Comment by mloubout** [comment]
Thanks for the questions, let me try to give you some answers and pointers

1. JUDI does not make any assumption on receiver positions, you can have them at random elevation or anywhere in the model without any problem. As of reciprocity, this is up to you to transform your data if this is something you would like to do. But the FWI sensitivity does not have ever any assumption about receiver positions or elevation and neither does reciprocity. So you should be fine with whichever geometry you have
2. That gradient does not look very useful. I would take small step and first work with a single source and look at the gradient. There is a lot of details like source frequency, grid spacing, ... that can impact the quality of the gradient
3. Doing RTM on CIG, I don't thin I've seen people do that, the shot or receiver gathers are to be imaged
4. This looks like preporcessed data. As I discussed with @kerim371 , the turning waves and long offset response are muted which are the part of the data you want for FWI. So you way wanna see if you can get the data without the muting

You can check this discussion for more details on FWI/RTM on field data
https://github.com/slimgroup/JUDI.jl/discussions/160
https://github.com/slimgroup/JUDI.jl/discussions/146

**Reply by zhangxiaoshuotttt** [reply]
Thank you very much for your suggestion. I will check it carefully

**Comment by kerim371** [comment]
Just to add few things:
1) as @mloubout your data is preprocessed for imaging, i.e. there is no turning (direct) waves. FWI result is best when working with turning waves. For such cases JUDI has mutings: [DataMute](https://slimgroup.github.io/JUDI.jl/dev/preconditioners/#Data-topmute) and [TopMute](https://slimgroup.github.io/JUDI.jl/dev/preconditioners/#Water-column-muting-(top-mute)). DataMute is a linear (triangle) muter based on `t0` and velocity `vp`. Thus usually your SEGY file should better contain both: reflected and turning waves. You mute reflections or turnings waves with JUDI data muter operator. TopMute is to mute water layer in model to have cleaner RTM/LSRTM at the top. To mute FWI near offsets high amplitudes we usually mute the gradient (there are examples in the repo). So for FWI I usually have `fval, gradient = fwi_objective(model, q, Ml*dobs, options=opt)` where `Ml` is DataMute operator. Then `gradient` should be muted at the top.
2) your RTM images are very strange. Even with pretty incorrect velocity model your RTM should have some coherent reflections seen. But here is only noise seen. Dont' you forget to set `options.IC = "fwi"` for FWI and `options.IC = "isic"` for RTM/LSRTM?
3) control the gradient at each iteration (save it as a picture). Do you work with the data that contains multiples or did you supress the multiples in your SEGY? if your data contains multiples you probably should better set `options.free_surface=true`.
4) what is the source/receiver spacings and how much channels per shot? and how much of shots?

Also the signature of source wavelet is important. And the low frequencies of your data.
